pkg/controllers/user/ingress/ingress_common.go

```diff
@@ -36,3 +42,77 @@ func GetIngressState(obj *v1beta1.Ingress) map[string]string {
 	}
 	return nil
 }
+
+func needNodePort(cluster *config.UserContext) bool {
+	c, err := cluster.Management.Management.Clusters("").Get(cluster.ClusterName, metav1.GetOptions{})
```

Comment:
lets not pass the context to util method. Fetch the cluster in the caller stack, and pass it here. Also use cluster lister, no direct api calls